frontend/coprs_frontend/coprs/auth.py

```diff
@@ -31,11 +31,16 @@ def next_url(cls):
         """
         if url := flask.request.values.get("next"):
             if cls._is_safe_next_url(url):
-                return flask.request.values.get("next")
+                return url
 
         if url := flask.request.referrer:
             if cls._is_safe_next_url(url):
-                return flask.request.referrer
+                return url
+
+        # When reading the URL from a cookie, pop it so it's usable only once
+        if url := flask.session.pop("next", None):
+            if cls._is_safe_next_url(url):
+                return url
```

Comment:
Shouldn't we move this logic above the referrer logic?  (meaning that, even if referrer is set, we should redirect per "next"?)